### PR TITLE
Implemented generalized phylogenetic diversity

### PR DESCRIPTION
**Pull request by qiyunzhu**
Phylogenetic diversity (PD) is a common measure of alpha diversity. In addition to the classical Faith's PD, there are multiple PD variants, some of which take feature abundance into consideration. This may be useful in microbiome research, especially shotgun metagenomics, where there could be prevalent low-abundance false-positive assignments. It was shown in a previous work ([Zhu et al., 2022](https://journals.asm.org/doi/full/10.1128/msystems.00167-22)) that abundance-weighted beta diversity metrics (Bray-Curtis and weighted UniFrac) generate much more stable results than unweighted ones at a gradient of sampling depths. The same characteristic for alpha diversity is worth exploring. A summary of these metrics is provided in [McCoy and Matsen, 2013](https://peerj.com/articles/157/). This implementation as well as the extensive addition of documentation is a result of my literature review efforts.

@wasade Most added content is documentation and unit test instead of algorithm code. The actual algorithm is very simple (see `skbio.diversity.alpha._pd._phydiv`). GitHub marked a large number of changed lines because I renamed `_faith_pd.py` as `_pd.py`.

Please complete the following checklist:

* [x] I have read the guidelines in [CONTRIBUTING.md](https://github.com/biocore/scikit-bio/blob/master/.github/CONTRIBUTING.md).

* [x] I have documented all public-facing changes in [CHANGELOG.md](https://github.com/biocore/scikit-bio/blob/master/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/biocore/scikit-bio/blob/master/LICENSE.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied. **It is your responsibility to disclose code, documentation, or other content derived from external source(s).** If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [ ] **This pull request does not include code, documentation, or other content derived from external source(s).**

**Note:** [REVIEWING.md](https://github.com/biocore/scikit-bio/blob/master/REVIEWING.md) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

**Comment by wasade** [comment]
Hi @qiyunzhu, it looks like there may be a few commits here which are unrelated to the PR. Would it be possible to rebase?

**Comment by qiyunzhu** [comment]
@wasade Thanks for noting this. I am quite newbie at this. I did `git rebase master` in my `pd` branch but nothing happens. Is there a specific set of commands you would suggest me to run?

**Comment by qiyunzhu** [comment]
@mataton Hello Matt, would you like to try to review this PR? Just follow the previous examples. Once you are done, ping me and @mortonjt . Thank you!

**Review by mataton — APPROVED** [review]
@qiyunzhu @mortonjt This looks good to me.

**Comment by mortonjt** [comment]
LGTM!

File: skbio/diversity/alpha/_pd.py [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+    :math:`\theta=0.25` or :math:`0.5` [4]_.
+
+    The parameter :math:`\theta` is analogous to the parameter :math:`\alpha`
+    in the generalized UniFrac metric [7]_.
+
+    Likewise, the rooted version of balance-weighted phylogenetic diversity,
+    :math:`RBWPD_{\theta}` [4]_ (although "balance" is not involved), is
+    defined as:
+
+        PD = \sum_{b \in T \sqcup R} l(b) p(b)^\theta
```
Comment (mataton) [inline skbio/diversity/alpha/_pd.py:348]: @qiyunzhu @mortonjt Apologies for the late catch, but I built the docs locally and line 348 does not render correctly.

**Comment by mortonjt** [comment]
That’s OK. Just submit another pull request to fix

**Pull request merged**